src/components/Toast/src/Toast.vue

```diff
@@ -0,0 +1,316 @@
+<template>
+	<m-bread
+		:class="[
+			$s.Toast,
+			{
+				[$s.EnhanceContrast]: enhanceContrast,
+			},
+		]"
+		:persistent="resolvedPersistent"
+		:dismiss-after="resolvedDismissAfter"
+		:style="style"
+		v-bind="$attrs"
+		v-on="$listeners"
+	>
+		<div :class="$s.ToastContent">
+			<m-icon
+				v-if="!resolvedHideIcon"
```

Comment:
Is `hideIcon` as a separate prop necessary? Could we conditionally render it or hide if depending on if an `icon` prop or slot content is present?

Reply:
it's necessary. making `undefined`, `null`, or `'' (empty string)` valid values for `iconName` is not good for the following reasons:
- `undefined` is indistinguishable from actually NOT setting the value, and `resolveThemeableProps` will look further up the theming hierarchy to find a valid value
- `null`, while distinguishable from not setting the value, is currently caught by `resolveThemeableProps` similarly to `undefined` and changing this would be a breaking change, so i'd prefer not to make this change, furthermore, I don't like allowing `null` as a valid value for `iconName` since semantically `null` is not a valid icon name, also the `iconName` is used across several other components and not all of these components have optional icons, some do and some don't, so it'd be weird if some components supported `null` values to hide the icons and others don't
- `'' (empty string)` is not good for the same reason as `null`: it's not semantic and it wouldn't be consistent with other components which use the `iconName` prop

`hideIcon` is clear, semantic, keeps behavior & valid values consistent between components which have mandatory icon slots vs optional icon slots. furthermore, it allows us to both set a default icon and a default visibility status for the default icon in each pattern, so altho an icon might be hidden by default, if the user chooses they would like to show the icon then the correct icon will be shown, which is not possible if we allow null-ish/falsey values for `iconName`, e.g.:

this is better:
```js
{
  "iconName": "info",
  "hideIcon": false, // icon might be hidden by default BUT if they set this to true they will get the correct icon for the context/pattern
}
```

than this:
```js
{
  "iconName": "", // okay, icon is hidden but... if I want to show it what should I change this to!?!?
}
```

Reply:
Actually, since the default value for `hideIcon` is `true` that means whenever they want to show icons they have to pass `<m-toast :hide-icon="false" />` where if we change the prop name to `shownIcon` and set the default value to `false` if they want to show the icon they only have to do `<m-toast show-icon />` so i've refactored the code and changed `hideIcon` to `showIcon`.